### PR TITLE
Support barrier mode for mapInPandas/mapInArrow [databricks]

### PR DESCRIPTION
**Pull request by wbo4958**
To fix https://github.com/NVIDIA/spark-rapids/issues/10344

Spark 3.5 has introduced a new feature supporting barrier mode for mapInPandas/mapInArrow, more detail can be found at https://issues.apache.org/jira/browse/SPARK-42896. However, spark-rapids missed this feature which resulted in unexpected behavior. For example

``` python
spark.range(10).mapInPandas(lambda x: x, "id long", True)
```

The same tasks of the above code will run on barrier mode on the CPU, while on `non-barrier` mode on the GPU with spark-rapids.

### CODE REVIEW

**Comment by wbo4958** [comment]
build

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala
Comment (jlowe) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala:1]: 2024 copyrights

File: sql-plugin/src/main/spark350/scala/com/nvidia/spark/rapids/shims/PythonMapInArrowExecShims.scala
Comment (jlowe) [inline sql-plugin/src/main/spark350/scala/com/nvidia/spark/rapids/shims/PythonMapInArrowExecShims.scala:1]: 2024 copyrights

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala
Comment (wbo4958) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuMapInBatchExec.scala:1]: Done, Thx

**Review by jlowe — APPROVED** [review]

**Pull request merged**